invisible_cities/io/mcinfo_io.py

```diff
@@ -209,16 +209,20 @@ def read_mcinfo_evt (mctables: (tb.Table, tb.Table, tb.Table, tb.Table),
                 previous_row = h5extents[iext-1]
 
                 ihit         = int(previous_row['last_hit']) + 1
-                ipart        = int(previous_row['last_particle']) + 1
+                if not return_only_hits:
+                    ipart        = int(previous_row['last_particle']) + 1
 
             ihit_end  = this_row['last_hit']
-            ipart_end = this_row['last_particle']
+            if not return_only_hits:
```

Comment:
Cant we lower  `ipart_end = this_row['last_particle']`  and put it after `if return_only_hits: break`

Reply:
Yes.